cmake/InstallDebugSymbols.cmake

```
@@ -0,0 +1,85 @@
+            message(SEND_ERROR "install_debug_symbols call missing required DESTINATION argument")
+            return()
+        endif()
+        if(NOT INSTALLSYMS_TARGETS)
+            message(SEND_ERROR "install_debug_symbols call missing required TARGETS argument")
+            return()
+        endif()
+        if(NOT INSTALLSYMS_CONFIGURATIONS)
+            set(INSTALLSYMS_CONFIGURATIONS RelWithDebInfo Debug)
+        endif()
+
+        # Wrap each config in a generator expression
+        set(HAS_SYMBOLS_CONDITION)
+        foreach(_config ${INSTALLSYMS_CONFIGURATIONS})
+            list(APPEND HAS_SYMBOLS_CONDITION "$<CONFIG:${_config}>")
+        endif()
+        # make list comma separated
+        string(REPLACE ";" "," HAS_SYMBOLS_CONDITION "${HAS_SYMBOLS_CONDITION}")
+        # Wrap in an "OR" generator expression
+        set(HAS_SYMBOLS_CONDITION "$<OR:${HAS_SYMBOLS_CONDITION}>")
+
+        set(EXTRA_INSTALL_ARGS ${INSTALLSYMS_PASSTHRU} ${INSTALLSYMS_UNPARSED_ARGUMENTS})
+        if(INSTALLSYMS_DESTINATION)
+            set(DEST ${INSTALLSYMS_DESTINATION})
+        endif()
+        foreach(_target ${INSTALLSYMS_TARGETS})
+            if(NOT INSTALLSYMS_DESTINATION)
+                get_target_property(_target_type ${_target} TYPE)
+                if("${_target_type}" STREQUAL "EXECUTABLE" OR "${_target_type}" STREQUAL "SHARED_LIBRARY")
+                    # exe or dll: put it alongside the runtime component
+                    set(DEST ${CMAKE_INSTALL_BINDIR})
+                else()
+                    set(DEST ${CMAKE_INSTALL_LIBDIR})
+                endif()
+            endif()
+            install(FILES $<${HAS_SYMBOLS_CONDITION}:$<TARGET_PDB_FILE:${_target}>>
+                DESTINATION ${DEST}
+                ${EXTRA_INSTALL_ARGS})
+        endforeach()
+    endfunction()
+else()
+    function(install_debug_symbols)
+        # do nothing if too old of CMake or not MSVC.
```

Perhaps emit a dev warning if CMake is too old so it doesn't silently fail?